src/nvim/ui_compositor.c

```diff
@@ -101,6 +101,32 @@ bool ui_comp_should_draw(void)
   return composed_uis != 0 && valid_screen;
 }
 
+// Raises or lowers the layer.
+//
+// @param[in]  layer_idx Index of the layer to be raised/lowered.
+// @param[in]  raise     Raise the layer if true, else lower it.
+void ui_comp_layers_adjust(size_t layer_idx, bool raise)
```

Comment:
what is the primary purpose of this function? is its main purpose to "sync" zindex with comp_index? or is that just a detail, and the main purpose is to raise/lower a layer?

how does it different from `ui_comp_raise_grid` ? (why aren't we using `ui_comp_raise_grid`?)

the answers to the above will help lead to a better name for this function. "adjust" is just meaningless.

Reply:
new implementation mainly avoids the use of full qsort, updates the order of the window grid in the layers according to the change of zindex, and ensures the correct order when the windows are drawn.

Reply:
> updates the order of the window grid in the layers according to the change of zindex, and ensures the correct order when the windows are drawn.

Ok that sounds like "syncs comp_index with zindex". Right? 

Is the *primary* purpose of the function to raise/lower, or to "sync" comp_index? Please update the docstring and function name, based on that.

Reply:
yes.